testplan/testing/multitest/base.py

```diff
@@ -988,17 +1008,22 @@ def _run_parallel_testcases(self, testsuite, execution_groups):
                 # If any testcase errors and we are configured to stop on
                 # errors, we still wait for the rest of the current execution
                 # group to finish before stopping.
-                if (
-                    testcase_report.status == Status.ERROR
-                    and self.cfg.stop_on_error
-                ):
-                    should_stop = True
+                if testcase_report.status == Status.ERROR:
```

Comment:
The original _run_parallel_testcases implementation looks problematic to me. How do we know the futures have all finished running before we do future.result().

Reply:
future.result waits for completion as per my local check. So it is functioning as it is proper and you can see the other cases executing in the background.

Reply:
I see, i was wrong in my previous comment.